File: pyNCAA/scripts/_check_grading.py
#!/usr/bin/env python3
# scripts/_check_grading.py — Trace grading logic on specific games
import json

store = json.loads(open("data/history.json").read())
runs = store.get("runs") or []

for r in runs:
    for g in (r.get("games") or []):
        if r.get("date") == "20260102" and g.get("away") and "Michigan" in g["away"] and g.get("home") and "USC" in g["home"]:
            print("Found game:", json.dumps(g, indent=2))
            print("\n--- Grading analysis ---")
            print(f"away: {g['away']} | home: {g['home']}")
            print(f"awayScore: {g.get('awayScore')} | homeScore: {g.get('homeScore')}")
            print(f"line: {g.get('line')} (positive = home is dog)")
            abs_line = abs(g.get("line", 0))
            home_dog = g.get("line", 0) > 0
            print(f"homeDog: {home_dog} | absLine: {abs_line}")
            if home_dog:
                margin = g["homeScore"] - g["awayScore"]
                v = margin + g["line"]
                print(f"Home margin: {margin} + line: {g['line']} = spreadResult: {v}")
                print(f"Home covers? {v > 0}")
            else:
                margin = g["homeScore"] - g["awayScore"]
                v = margin + g["line"]
                print(f"Home margin: {margin} + line: {g['line']} = v: {v}")
                print(f"Away covers? {v < 0}")

print("\n\n=== VERIFY WITH ACTUAL GRADED PICKS ===")
checked = 0
for r in runs:
    for g in (r.get("games") or []):
        if checked >= 5:
            break
        if not g.get("sPick") or g["sPick"] == "PASS":
            continue
        if not g.get("result"):
            continue
        print(f"\n{r.get('date')}: {g['away']} @ {g['home']}")
        print(f"  sPick: {g['sPick']} | result: {g['result']}")
        print(f"  line: {g.get('line')} | score: {g.get('awayScore')}-{g.get('homeScore')}")
        print(f"  pHomeCover: {g.get('pHomeCover')} | pAwayCover: {g.get('pAwayCover')}")
        checked += 1

File: jsNCAA/scripts/_check_grading.mjs
#!/usr/bin/env node
import fs from "fs";

const store = JSON.parse(fs.readFileSync("data/history.json", "utf8"));
const runs = store.runs || [];

// Pick a known game and trace the grading
// Michigan +22.5 vs USC: away=Michigan(66), home=USC... wait, let me find the actual record
for (const r of runs) {
  for (const g of r.games) {
    if (r.date === "20260102" && g.away && g.away.includes("Michigan") && g.home && g.home.includes("USC")) {
      console.log("Found game:", JSON.stringify(g, null, 2));
      console.log("\n--- Grading analysis ---");
      console.log("away:", g.away, "| home:", g.home);
      console.log("awayScore:", g.awayScore, "| homeScore:", g.homeScore);
      console.log("line:", g.line, "(negative = home favored, positive = home is dog)");

      const absLine = Math.abs(g.line);
      const homeDog = g.line > 0;
      console.log("homeDog:", homeDog, "| absLine:", absLine);

      if (homeDog) {
        // Home is getting points. Home covers if homeScore + line > awayScore
        // i.e. (homeScore - awayScore) + line > 0
        const margin = g.homeScore - g.awayScore;
        const v = margin + g.line;
        console.log("Home margin:", margin, "+ line:", g.line, "= spreadResult:", v);
        console.log("Home covers?", v > 0);
      } else {
        // Away is getting points. Away covers if awayScore + absLine > homeScore
        // i.e. (awayScore - homeScore) + absLine > 0
        // or: -(homeScore - awayScore) - line > 0 (since line < 0)
        // or: -(margin + line) > 0
        const margin = g.homeScore - g.awayScore;
        const v = margin + g.line;
        console.log("Home margin:", margin, "+ line:", g.line, "= v:", v);
        console.log("Away covers?", v < 0);
      }
    }
  }
}

// Also check: what does the line sign convention actually mean?
// Let's look at games where sPick was made and result is known
console.log("\n\n=== VERIFY WITH ACTUAL GRADED PICKS ===");
let checked = 0;
for (const r of runs) {
  for (const g of r.games) {
    if (checked >= 5) break;
    if (!g.sPick || g.sPick === "PASS") continue;
    if (!g.result) continue;
    console.log(`\n${r.date}: ${g.away} @ ${g.home}`);
    console.log(`  sPick: ${g.sPick} | result: ${g.result}`);
    console.log(`  line: ${g.line} | score: ${g.awayScore}-${g.homeScore}`);
    console.log(`  pHomeCover: ${g.pHomeCover} | pAwayCover: ${g.pAwayCover}`);
    checked++;
  }
}
